### PR TITLE
Release 2 verified update docs

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)

Updates docs links from release_2_docs to release_2_verified_docs tag.
Update python installation instructions in package doc.
Add explicit version in install docs
Add version check and warn if outside supported range.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [x] Documentation update
- [ ] Other (please describe)

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Runtime/Communicator/RpcCommunicator.cs [hunk 1]
```diff
@@ -96,6 +101,27 @@ public RpcCommunicator(CommunicatorInitParameters communicatorInitParameters)
...
         }
 
+        internal static bool CheckPythonPackageVersionIsCompatible(string pythonLibraryVersion)
+        {
+            Version pythonVersion;
+            try
+            {
+                pythonVersion = new Version(pythonLibraryVersion);
+            }
+            catch
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Communicator/RpcCommunicator.cs:111]: I don't love not specifying the exception type here, but there are a bunch of exception type that parsing the version can throw, and we'd handle all of them the same way anyway.

File: docs/Installation.md [hunk 2]
```diff
@@ -123,7 +123,7 @@ To install the `mlagents` Python package, activate your virtual environment and
 run from the command line:
 
 ```sh
-pip3 install mlagents
+pip3 install mlagents==0.16.1
```
Comment (vincentpierre) [inline docs/Installation.md:126]: Not sure if this document is supported but `ml-agents/docs/Installation-Anaconda-Windows.md` also mentions installing mlagents and maybe needs a fixed version as well...

File: com.unity.ml-agents/Tests/Editor/Communicator/RpcCommunicatorTests.cs [hunk 3]
```diff
@@ -50,5 +50,21 @@ public void TestCheckCommunicationVersionsAreCompatible()
...
+            Assert.IsFalse(RpcCommunicator.CheckPythonPackageVersionIsCompatible("0.42.0")); // too high
+
+            // These are fine
+            Assert.IsTrue(RpcCommunicator.CheckPythonPackageVersionIsCompatible("0.16.1"));
+            Assert.IsTrue(RpcCommunicator.CheckPythonPackageVersionIsCompatible("0.17.17"));
+            Assert.IsTrue(RpcCommunicator.CheckPythonPackageVersionIsCompatible("0.20.0"));
+
+            // "dev" string or otherwise unparseable
+            Assert.IsFalse(RpcCommunicator.CheckPythonPackageVersionIsCompatible("0.17.0-dev0"));
+            Assert.IsFalse(RpcCommunicator.CheckPythonPackageVersionIsCompatible("oh point seventeen point oh"));
```
Comment (vincentpierre) [inline com.unity.ml-agents/Tests/Editor/Communicator/RpcCommunicatorTests.cs:67]: 🤣

**Review by vincentpierre — APPROVED** [review]

File: docs/Installation.md (hunk 2 above)
Comment (chriselion) [reply docs/Installation.md:126]: Good catch, updated that too.

**Pull request merged**